pkg/client/client.go

```diff
@@ -180,77 +209,191 @@ func (o *OptimizelyClient) GetFeatureVariableBoolean(featureKey, variableKey str
 
-	value, valueType, err := o.GetFeatureVariable(featureKey, variableKey, userContext)
+	value, valueType, featureDecision, err := o.getFeatureVariable(featureKey, variableKey, userContext)
+
+	defer func() {
+		if o.notificationCenter != nil {
+			variableMap := map[string]interface{}{
+				"variableKey":   variableKey,
+				"variableType":  valueType,
+				"variableValue": value,
+			}
+
+			decisionNotification := decision.FeatureNotificationWithVariables(featureKey, featureDecision, &userContext, variableMap)
+			decisionNotification.Type = notification.FeatureVariable
+
+			if e := o.notificationCenter.Send(notification.Decision, *decisionNotification); e != nil {
+				o.logger.Warning("Problem with sending notification")
+			}
+		}
+	}()
 	if err != nil {
 		return "", err
 	}
 	if valueType != entities.String {
 		return "", fmt.Errorf("variable value for key %s is wrong type", variableKey)
 	}
+
 	return value, err
 }
 
 // GetFeatureVariableJSON returns the feature variable value of type json associated with the given feature and variable keys.
 func (o *OptimizelyClient) GetFeatureVariableJSON(featureKey, variableKey string, userContext entities.UserContext) (value *optimizelyjson.OptimizelyJSON, err error) {
 
-	val, valueType, err := o.GetFeatureVariable(featureKey, variableKey, userContext)
+	val, valueType, featureDecision, err := o.getFeatureVariable(featureKey, variableKey, userContext)
+	defer func() {
+		if o.notificationCenter != nil {
+			var variableValue interface{}
+			if value != nil {
+				variableValue = value.ToMap()
+			} else {
+				variableValue = val
+			}
+			variableMap := map[string]interface{}{
+				"variableKey":   variableKey,
+				"variableType":  valueType,
+				"variableValue": variableValue,
+			}
+			decisionNotification := decision.FeatureNotificationWithVariables(featureKey, featureDecision, &userContext, variableMap)
+			decisionNotification.Type = notification.FeatureVariable
+
+			if e := o.notificationCenter.Send(notification.Decision, *decisionNotification); e != nil {
+				o.logger.Warning("Problem with sending notification")
+			}
+		}
+	}()
 	if err != nil {
 		return value, err
 	}
 
 	value, err = optimizelyjson.NewOptimizelyJSONfromString(val)
 	if err != nil || valueType != entities.JSON {
-		return nil, fmt.Errorf("variable value for key %s is invalid or wrong type", variableKey)
+		value, err = nil, fmt.Errorf("variable value for key %s is invalid or wrong type", variableKey)
 	}
 
 	return value, err
 }
 
 // GetFeatureVariable returns feature variable as a string along with it's associated type.
-func (o *OptimizelyClient) GetFeatureVariable(featureKey, variableKey string, userContext entities.UserContext) (value string, valueType entities.VariableType, err error) {
+func (o *OptimizelyClient) getFeatureVariable(featureKey, variableKey string, userContext entities.UserContext) (string, entities.VariableType, *decision.FeatureDecision, error) {
```

Comment:
Nit: Any more descriptive names we can use, rather than having `getFeatureVariable` and `GetFeatureVariable`? If not changing, the doc comment on this one needs fixing (capitalized).

Reply:
yeah, that was my bad, the comment should not belong here, thanks for catching. GetFeatureVariable has to stay here, since this is a public API. getFeatureVariable was done only for notification support - to separate this notification and not to have double calls. I did not want to add anything to this function name like getFeatureVariableWithNotification or getFeatureVariableHelper, but any suggestions are welcome.

Reply:
I don't have any suggestions, but will let you know if I think of one.